### PR TITLE
Fix/bridge

### PR DESCRIPTION
**Pull request by danielsimao**
Closes # 

## 📝 Description

- Remove optimism sdk
- remove old bob chain defenition
-

### CODE REVIEW

File: apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
+/* eslint-disable no-console */
```
Comment (danielsimao) [inline apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx:1]: main file to review

File: apps/evm/src/hooks/tests/useWalletClient.test.ts [hunk 2]
```diff
@@ -10,20 +11,37 @@ import { chainL1, chainL2 } from '@/constants';
...
   walletActionsL1: vi.fn(),
   walletActionsL2: vi.fn()
 }));
 
+vi.mock(import('usehooks-ts'), async (importOriginal) => {
+  const actual = await importOriginal();
+
+  return {
+    ...actual,
+    useQueryClient: vi.fn()
```
Comment (slavastartsev) [inline apps/evm/src/hooks/tests/useWalletClient.test.ts:35]: I think it's a typo -- `useQueryClient` comes from different package

File: apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx [hunk 3]
```diff
@@ -309,120 +273,99 @@ const BobBridgeForm = ({
...
-
   useEffect(() => {
     form.resetForm();
-    gasEstimateMutation.reset();
 
     setTicker(nativeToken.symbol);
     setGasTicker(nativeToken.symbol);
     setAmount('');
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [direction]);
```
Comment (slavastartsev) [inline apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx:366]: this part could be rewriten
```
  const [prevDirection, setPrevDirection] = useState<TransactionDirection>();

  if (prevDirection !== direction) {
    setPrevDirection(direction);

    form.resetForm();
    gasEstimateMutation.reset();

    setTicker(nativeToken.symbol);
    setGasTicker(nativeToken.symbol);
    setAmount('');
  }
```

this solves the issue of exhaustive deps + less rerenders

File: apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx [hunk 4]
```diff
@@ -105,111 +116,62 @@ const BobBridgeForm = ({
...
-      }
-
-      const gasAmount = await (direction === TransactionDirection.L1_TO_L2
-        ? messenger.estimateGas.depositERC20(l1Address, l2Address, amount)
-        : messenger.estimateGas.withdrawERC20(l1Address, l2Address, amount));
-
-      return CurrencyAmount.fromRawAmount(nativeToken, gasAmount.toBigInt());
-    }
-  });
+  const handleSuccess = useCallback(
```
Comment (slavastartsev) [inline apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx:133]: I think it's goind to be fine if we do not memoize `handleError` and `handle Success` -> `onBridgeSuccess` and `onFailBridge` props are not memoized so `useCallback` effectively does nothing

**Review by slavastartsev — APPROVED** [review]
LGTM. I assume this code was tested cause I see pending txs on bridge page.

File: apps/evm/src/hooks/tests/useWalletClient.test.ts [hunk 5]
```diff
@@ -10,20 +11,37 @@ import { chainL1, chainL2 } from '@/constants';
...
+
+  return {
+    ...actual,
+    useIsClient: vi.fn()
+  };
+});
+
+vi.mock('usehooks-ts', () => ({
+  useIsClient: vi.fn()
+}));
```
Comment (slavastartsev) [inline apps/evm/src/hooks/tests/useWalletClient.test.ts:41]: this mock and `vi.mock` above do the same thing.
could be removed

Comment (danielsimao) [reply apps/evm/src/hooks/tests/useWalletClient.test.ts:41]: Sorry...got confused xD removed

File: apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx [hunk 6]
```diff
@@ -473,19 +402,21 @@ const BobBridgeForm = ({
     hideErrors: 'untouched'
   });
 
-  const handleChangeTicker = (currency: Currency) => {
-    setTicker(currency.symbol as string);
-
-    const selectedToken = tokens?.find((token) => token.l1Currency.symbol === currency.symbol);
+  const handleReset = () => {
+    setPrevDirection(direction);
```
Comment (slavastartsev) [inline apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx:406]: let's move setPrevDirection(direction); out of handleReset function -- feels like it has nothing to do with resetting values but just updates prevDirection

File: apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx [hunk 7]
```diff
@@ -473,19 +402,21 @@ const BobBridgeForm = ({
...
 
-    if (!selectedCurrency || !selectedToken) return;
+    setTicker(defaultTicker);
+    setAmount('');
+  };
 
-    const currencyAmount = CurrencyAmount.fromBaseAmount(selectedCurrency, amount || 0);
+  if (prevDirection !== direction) {
+    handleReset();
+  }
```
Comment (slavastartsev) [inline apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx:416]:
```suggestion
  if (prevDirection !== direction) {
    setPrevDirection(direction);
    handleReset();
  }
```

File: apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx [hunk 8]
```diff
@@ -92,124 +110,64 @@ const BobBridgeForm = ({
...
-      const gasAmount = await (direction === TransactionDirection.L1_TO_L2
-        ? messenger.estimateGas.depositERC20(l1Address, l2Address, amount)
-        : messenger.estimateGas.withdrawERC20(l1Address, l2Address, amount));
+    handleReset();
 
-      return CurrencyAmount.fromRawAmount(nativeToken, gasAmount.toBigInt());
-    }
-  });
+    refetchBalances();
+  };
```
Comment (slavastartsev) [inline apps/evm/src/app/[lang]/(bridge)/bridge/components/BridgeForm/BobBridgeForm.tsx:138]:
```suggestion
  const handleSuccess = (data: BridgeTransaction) => {
    onBridgeSuccess?.(data);

    setPrevDirection(direction);
    handleReset();

    refetchBalances();
  };

```

**Pull request merged**